File: backend/internal/v2/dto/staking.go
package dto

type StakingPoolResponse struct {
	PoolID             int     `json:"pool_id"`
	Address            string  `json:"address"`
	StakingToken       string  `json:"staking_token"`
	StakingTokenSymbol string  `json:"staking_token_symbol"`
	RewardToken        string  `json:"reward_token"`
	RewardTokenSymbol  string  `json:"reward_token_symbol"`
	TotalStaked        string  `json:"total_staked"`
	RewardRate         string  `json:"reward_rate"`
	APR                float64 `json:"apr"`
	UserStaked         string  `json:"user_staked"`
	UserPendingRewards string  `json:"user_pending_rewards"`
	IsClosed           bool    `json:"is_closed"`
}

File: frontend/src/types/staking.ts
export interface StakingPool {
  pool_id: number;
  address: string;
  rewarder_address: string;
  staking_token: string;
  staking_token_symbol: string;
  reward_token: string;
  reward_token_symbol: string;
  total_staked: string;
  reward_rate: string;
  apr: number;
  user_staked: string;
  user_pending_rewards: string;
  total_reward_pool: string;
  remaining_rewards: string;
  reward_token_decimals: number;
  is_closed: boolean;
}

export interface StakingAction {
  pool_id: number;
  pool_address: string;
  amount: string;
}
